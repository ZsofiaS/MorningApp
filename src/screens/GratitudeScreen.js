import React, { useState, useEffect } from 'react';
import { TouchableOpacity, View, Text, StyleSheet, TextInput, FlatList, ScrollView } from 'react-native';
import { Input, Button } from 'react-native-elements';
import colours from '../../constants/colours.js';
import { init, insertEntry, fetchEntries, deleteEntry } from '../../helpers/db';
import Moment from 'react-moment';
import Icon from 'react-native-vector-icons/FontAwesome';

init().then(() => {
  console.log('Initialised database')
}).catch(err => {
  console.log('Initialising database failed');
  console.log(err);
});

export default function GratitudeScreen() {
  const [entry, setEntry] = useState('');
  const [entryList, setEntryList] = useState([]);

  const saveEntry = async () => {
    setEntry(entry);
    const dbResult = await insertEntry(entry, Date.now());
    setEntry('');
    showEntries();
  }

  const showEntries = async () => {
    const dbResult = await fetchEntries();
    setEntryList(dbResult.rows._array.reverse())
  }

  const convertDate = (date) => {
    return (
      <Moment element={Text} format='Do MMM YYYY HH:mm'>{date}</Moment>
      )
  }

  const removeEntry = async (id) => {
    const dbResult = await deleteEntry(id);
    showEntries();
  }

  useEffect(() => {
    showEntries();
  }, []);

  return (
    <ScrollView keyboardShouldPersistTaps='handled'>
      <Text
        style={styles.title}>
        What are you grateful for today?
      </Text>
      <TextInput
        multiline={true}
        numberOfLines={4}
        onChangeText = { value => setEntry(value)}
        value={entry}
        style={{paddingHorizontal: 50, fontSize: 18}}
        inputContainerStyle={styles.input}
        placeholder="A good night's sleep? Sunny weather?"
        />
      <View style={styles.buttonContainer}>
        <Button
          titleStyle={{ color: colours.primaryColour}}
          buttonStyle={{ padding: 15}}
          type='clear'
          title='Save journal entry'
          onPress={saveEntry}
          />
      </View>
      <ScrollView
        contentContainerStyle={styles.entryListContainer}>
      {
        entryList.map((item, i) => {
          return(
            <View
              key={i}
              style={styles.entryContainer}
              id={i}>
              <View style={styles.entryTitle}>
                <Text
                  style={styles.entryDate}>
                  {convertDate(item["date"])}
                </Text>
                <TouchableOpacity
                  onPress={() => removeEntry(item["id"])}
                >
                    <Icon
                      name='trash'
                      type='clear'
                      size={15}
                      color={colours.primaryColour}
                    />
                </TouchableOpacity>
              </View>
              <View
                style={styles.entryContentContainer}>
                <Text
                  style={styles.entryContent}>
                  {item["entry"]}
                </Text>

              </View>
            </View>
        )
        })
      }
    </ScrollView>
    </ScrollView>
  )
}

const styles = StyleSheet.create({
  buttonContainer: {
    width: '100%',
    alignItems: 'flex-end',
    paddingHorizontal: 50
  },
  title: {
    color: colours.primaryColour,
    fontSize: 20,
    marginVertical: 60,
    paddingHorizontal: 50
  },
  entryTitle: {
    flexDirection: 'row',
    justifyContent: 'space-between'
  },
  entryListContainer: {
    alignItems: 'flex-start'
  },
  entryContainer: {
    width: '100%',
    padding: 20,
  },
  entryContentContainer: {
    backgroundColor: colours.lightPurple,
    padding: 20,
    borderRadius: 5,
    marginVertical: 10
  },
  entryContent: {
    fontSize: 18
  },
  entryDate: {
    paddingLeft: 10
  }
})
