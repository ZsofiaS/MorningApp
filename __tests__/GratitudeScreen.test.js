import React from 'react';
import renderer from 'react-test-renderer';
import GratitudeScreen from '../src/screens/GratitudeScreen';

test('gratitudescreen renders correctly', () => {
  const tree = renderer.create(<GratitudeScreen />).toJSON();
  expect(tree).toMatchSnapshot();
})
