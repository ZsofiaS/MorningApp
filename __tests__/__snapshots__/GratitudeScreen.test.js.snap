// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gratitudescreen renders correctly 1`] = `
<RCTScrollView
  keyboardShouldPersistTaps="handled"
>
  <View>
    <Text
      style={
        Object {
          "color": "#c48b9f",
          "fontSize": 20,
          "marginVertical": 60,
          "paddingHorizontal": 50,
          "textAlign": "center",
        }
      }
    >
      What are you grateful for today?
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "marginVertical": 15,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 120,
            "width": 120,
          }
        }
      />
    </View>
    <TextInput
      allowFontScaling={true}
      multiline={true}
      numberOfLines={4}
      onChangeText={[Function]}
      placeholder="A good night's sleep? Sunny weather?"
      rejectResponderTermination={true}
      style={
        Object {
          "fontSize": 18,
          "paddingHorizontal": 50,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "paddingHorizontal": 50,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "borderRadius": 3,
            "overflow": "hidden",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "busy": false,
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderColor": "#2089dc",
                "borderRadius": 3,
                "borderWidth": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 15,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#c48b9f",
                  "fontSize": 18,
                  "paddingBottom": 1,
                  "paddingTop": 2,
                  "textAlign": "center",
                }
              }
            >
              Save journal entry
            </Text>
          </View>
        </View>
      </View>
    </View>
    <RCTScrollView
      contentContainerStyle={
        Object {
          "alignItems": "flex-start",
        }
      }
    >
      <View />
    </RCTScrollView>
  </View>
</RCTScrollView>
`;

exports[`homescreen renders correctly 1`] = `
<RCTScrollView
  keyboardShouldPersistTaps="handled"
>
  <View>
    <Text
      style={
        Object {
          "color": "#c48b9f",
          "fontSize": 20,
          "marginVertical": 60,
          "paddingHorizontal": 50,
          "textAlign": "center",
        }
      }
    >
      What are you grateful for today?
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "marginVertical": 15,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 120,
            "width": 120,
          }
        }
      />
    </View>
    <TextInput
      allowFontScaling={true}
      multiline={true}
      numberOfLines={4}
      onChangeText={[Function]}
      placeholder="A good night's sleep? Sunny weather?"
      rejectResponderTermination={true}
      style={
        Object {
          "fontSize": 18,
          "paddingHorizontal": 50,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "paddingHorizontal": 50,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "borderRadius": 3,
            "overflow": "hidden",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "busy": false,
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderColor": "#2089dc",
                "borderRadius": 3,
                "borderWidth": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 15,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#c48b9f",
                  "fontSize": 18,
                  "paddingBottom": 1,
                  "paddingTop": 2,
                  "textAlign": "center",
                }
              }
            >
              Save journal entry
            </Text>
          </View>
        </View>
      </View>
    </View>
    <RCTScrollView
      contentContainerStyle={
        Object {
          "alignItems": "flex-start",
        }
      }
    >
      <View />
    </RCTScrollView>
  </View>
</RCTScrollView>
`;
